API_MANIFESTO.md

```diff
@@ -0,0 +1,91 @@
+# The API Manifesto
+
+# Infinispan API dependency module
+
+The ```infinispan-api``` dependency will contain the public API. This module will be independent from the other modules. 
+Dependency with implementations will be **logical**. This will keep the modularity needed for the different use cases.
+
+Examples:
+ * Cache use case
+ 
+ * NoSQL data store use case
+ 
+ * Tasks, concurrency, counters uses cases
+
+The dependencies to thirty-party libraries must be avoided. This module should contain dependencies to JDK classes, 
+not more.
+
+# User First
+
+* Before making an API public, consider to test the usability with someone not familiar with the feature
+
+* Convention over Configuration: Make sure a default behaviour exists. Don't make me think.
+
+* Avoid exposing internals as much as possible. Even if your feature lays on top of a cache today, internals can change.
+
+* Each public API/Feature must have a simple tutorial showcasing the example. Without a demo, the API does not exist
+
+* If you are unsure about an API, mark it as ```@Experimental```. 
+
+* Don't let ```@Experimental``` tag last forever. 
+
+* Ask for user's feedback actively (zulip, user forums, blog post, twitter, github ...)
+
+* Be clear about whats PUBLIC API and what internals/implementations are. 
```

Comment:
Is it worth mentioning `impl` packages?

Reply:
Maybe saying packages should be called impl or something like that ?

Reply:
Yeah exactly. Anything isn't supported should be under `impl.*`

Reply:
done